modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceHandler.java

```diff
@@ -589,7 +589,21 @@ public void commitResponseHideExceptions(
             handleException("IO error submiting response : " + e.getMessage(), e, conn);
         }
     }
-    
+    /**
+     * Shutting down the thread pools.
+     */
+    public void stop() {
+        latencyView.destroy();
+        s2sLatencyView.destroy();
+        threadingView.destroy();
+        try {
+            if (sourceConfiguration.getWorkerPool() != null) {
+                sourceConfiguration.getWorkerPool().shutdown(1000);
+            }
+        } catch (InterruptedException e) {
+            log.warn("Error while shutting down worker thread pool. " + e.getMessage());
```

Comment:
shouldn't this be an error log? Also, don't we need to retry the operation if it was interrupted?

Reply:
Retry will not be successful. And this exception has to be ignored. That is why warn log is added.